#  IBM Data Science Capstone Project 
## Planning Travel and Tours in New York City's Broadway Theatre District
### Comparison of Hotels, Restaurants, and Theaters in Manhattan, NY for a Broadway Theater Trip
### Brian Vineyard 

### Business Problem


**Business Problem**: A travel service is planning to offer Broadway trips with the best options in New York City, NY to include on their package. They want to build a package with varying hotels, and a list of recommended restaurants. 

Their clients are mostly interested in Broadway musicals and plays, so this project will focus on finding the best hotels and restaurants near the Broadway theater district.

![Broadway](77.jpg)

**Key Questions to Answer**

- What are the best theaters to catch the top Broadway shows?
- What hotels close to the Theater District have the best ratings and room rates?
- What are the best nearby restaurants?


**Foursquare API study of following city**:

- Broadway Theatre District, New York City, NY

Example search criteria: Hotels, Restaurants, Museums, Live Shows

**Project Details**: 

This project uses the following tools and technology:

- [Foursquare Places API](https://https://enterprise.foursquare.com/products/places)
- [2014 New York City Neighborhood Names](https://geo.nyu.edu/catalog/nyu_2451_34572)
- [BeautifulSoup](https://beautiful-soup-4.readthedocs.io/en/latest/)


The goal is to find the best area of the city to hold tours, based on our findings.

**Data Sources and Description**:

This project will use data for the latitude and longitude of zipcodes located in the following city:

- New York, NY


The coordinates will be passed to the Foursquare Places API, for querying New York City neighborhoods for venues such as:

- **Restaurants**
    - Liebman's Deli, La Morada, Royal 35 Steakhouse, Le Benardin
- **Hotels**
    - Gramercy Park, Life Hotel Nomad, Mansfield Hotel
- **Fun**
    - Museums: Metropoliton Museum of Art, 9/11 Memorial, American Museum of Natural History
    - Top Tourist Destinations: Empire State Building, Madison Square Garden, Statue of Liberty, Times Square
    - Broadway Productions: Hamilton, The Lion King, Wicked, Moulin Rouge, Phantom of the Opera
    - Sightseeing Cruises: Circle Line Full Island, Classic Harbor Lines, Zephyr Yacht, Liberty Cruises
    - Parks: Central Park, Washington Square Park, Fort Tryon Park, Bronx Zoo, 
    
 ### Table of Contents
 1. [Import Libraries, Foursquare Credentials](#import-1)
 2. [New York's Manhattan Borough Neighborhood Data](#borough-2)
   - [Map of Neighborhoods in New York City's Five Boroughs](#borough-map)
   - [Map of Neighborhoods in NYC's Manhattan Borough](#neighor-map)
 3. [Broadway Theatres and Current Productions](#broadway-3)
   - [Map of Broadway Theatre Locations](#broadway-map)
 4. [Hotels Near the Broadway Theatre District](#hotels-4)
   - [Map of Hotels Near Broadway Theatres](#hotels-map)
 5. [Restaurants Near the Broadway Theatre District](#restaurants-5)
   - [Map of Restaurants Near Broadway Theatres](#rest_map)
 6. [Conclusions and Findings](#conclusions)

## 1. Import Libraries, Foursquare Credentials, and Neighborhood Data <a name="import-1"></a>

### Import Libraries ###

In [284]:
import numpy as np # library to handle data in a vectorized manner

import pandas as pd # library for data analsysis
pd.set_option('display.max_columns', None)
pd.set_option('display.max_rows', None)

import json # library to handle JSON files
import pickle

from geopy.geocoders import Nominatim # Convert addresses into GPS coordinates
from bs4 import BeautifulSoup

import requests # library to handle requests

from pandas.io.json import json_normalize # transform JSON file into a Pandas dataframe
# Matplotlib and associated plotting modules
import matplotlib.cm as cm
import matplotlib.colors as colors
# import k-means from clustering stage
from sklearn.cluster import KMeans

import folium # map rendering library
print('Libraries imported.')

Libraries imported.


### Load Foursquare Parameters ###

In [285]:
# Loads the Foursquare credentials for logging in to the API

file = open('credentials2.p', 'rb')
credentials = pickle.load(file)
file.close()
CLIENT_ID = credentials['CLIENT_ID']
CLIENT_SECRET = credentials['CLIENT_SECRET']
VERSION = '20180604'
LIMIT = 30

## 2. New York's Manhattan Borough Neighborhood Data <a name="borough-2"></a>

### Import Neighborhood Data ###

New York City has a total of 5 boroughs and 306 neighborhoods. We will be concentrating on the Manhattan borough, where Broadway and the Theatre District are located. We will start with a larger dataset, however, with the latitude and longitude of each of NYC's neighborhoods.

We will pull in the data as a JSON file, then load that into a pandas dataframe for analysis.

#### Open  and explore the JSON file for New York City Neighborhood Names ####

In [286]:
with open('nyu-2451-34572-geojson.json') as json_data:
    newyork_data = json.load(json_data)

In [287]:
# View the data from the JSON file

newyork_data

{'type': 'FeatureCollection',
 'totalFeatures': 306,
 'features': [{'type': 'Feature',
   'id': 'nyu_2451_34572.1',
   'geometry': {'type': 'Point',
    'coordinates': [-73.84720052054902, 40.89470517661]},
   'geometry_name': 'geom',
   'properties': {'name': 'Wakefield',
    'stacked': 1,
    'annoline1': 'Wakefield',
    'annoline2': None,
    'annoline3': None,
    'annoangle': 0.0,
    'borough': 'Bronx',
    'bbox': [-73.84720052054902,
     40.89470517661,
     -73.84720052054902,
     40.89470517661]}},
  {'type': 'Feature',
   'id': 'nyu_2451_34572.2',
   'geometry': {'type': 'Point',
    'coordinates': [-73.82993910812398, 40.87429419303012]},
   'geometry_name': 'geom',
   'properties': {'name': 'Co-op City',
    'stacked': 2,
    'annoline1': 'Co-op',
    'annoline2': 'City',
    'annoline3': None,
    'annoangle': 0.0,
    'borough': 'Bronx',
    'bbox': [-73.82993910812398,
     40.87429419303012,
     -73.82993910812398,
     40.87429419303012]}},
  {'type': 'Feature',
 

Notice how all the relevant data is in the features key, which is basically a list of the neighborhoods. So, let's define a new variable that includes this data.

In [288]:
neighborhoods_data = newyork_data['features']

Let's take a look at the first item in this list.

In [289]:
neighborhoods_data[0]

{'type': 'Feature',
 'id': 'nyu_2451_34572.1',
 'geometry': {'type': 'Point',
  'coordinates': [-73.84720052054902, 40.89470517661]},
 'geometry_name': 'geom',
 'properties': {'name': 'Wakefield',
  'stacked': 1,
  'annoline1': 'Wakefield',
  'annoline2': None,
  'annoline3': None,
  'annoangle': 0.0,
  'borough': 'Bronx',
  'bbox': [-73.84720052054902,
   40.89470517661,
   -73.84720052054902,
   40.89470517661]}}

#### Tranform the data into a pandas dataframe ####

In [290]:
# define the dataframe columns
column_names = ['Borough', 'Neighborhood', 'Latitude', 'Longitude'] 

# instantiate the dataframe
neighborhoods = pd.DataFrame(columns=column_names)

Loop through the data and load the dataframe one row at a time

In [291]:
for data in neighborhoods_data:
    borough = neighborhood_name = data['properties']['borough'] 
    neighborhood_name = data['properties']['name']
        
    neighborhood_latlon = data['geometry']['coordinates']
    neighborhood_lat = neighborhood_latlon[1]
    neighborhood_lon = neighborhood_latlon[0]
    
    neighborhoods = neighborhoods.append({'Borough': borough,
                                          'Neighborhood': neighborhood_name,
                                          'Latitude': neighborhood_lat,
                                          'Longitude': neighborhood_lon}, ignore_index=True)

In [292]:
# Look at the first five rows of data
neighborhoods.head()

Borough Neighborhood   Latitude  Longitude
0  Bronx   Wakefield    40.894705 -73.847201
1  Bronx   Co-op City   40.874294 -73.829939
2  Bronx   Eastchester  40.887556 -73.827806
3  Bronx   Fieldston    40.895437 -73.905643
4  Bronx   Riverdale    40.890834 -73.912585

In [293]:
print('The dataframe has {} boroughs and {} neighborhoods.'.format(
        len(neighborhoods['Borough'].unique()),
        neighborhoods.shape[0]
    )
)

The dataframe has 5 boroughs and 306 neighborhoods.


###  Map of Neighborhoods in the New York City' Five Boroughs <a name="borough-map"></a>

#### Get the New York City latitude and longitude coordinates from geopy ####

In [294]:
address = 'New York City, NY'

geolocator = Nominatim(user_agent="ny_explorer")
location = geolocator.geocode(address)
latitude = location.latitude
longitude = location.longitude
print('The geograpical coordinates of New York City are {}, {}.'.format(latitude, longitude))

The geograpical coordinates of New York City are 40.7127281, -74.0060152.


**Build map of New York City in Folium, with neighborhoods in the five boroughs indicated with blue markers**

In [295]:
# create map of New York using latitude and longitude values
map_newyork = folium.Map(location=[latitude, longitude], zoom_start=10)

# add markers to map
for lat, lng, borough, neighborhood in zip(neighborhoods['Latitude'], neighborhoods['Longitude'], neighborhoods['Borough'], neighborhoods['Neighborhood']):
    label = '{}, {}'.format(neighborhood, borough)
    label = folium.Popup(label, parse_html=True)
    folium.CircleMarker(
        [lat, lng],
        radius=5,
        popup=label,
        color='blue',
        fill=True,
        fill_color='#3186cc',
        fill_opacity=0.7,
        parse_html=False).add_to(map_newyork)  
    
map_newyork

**Map of New York City Neighborhoods in Five Boroughs: Manhattan, The Bronx, Brooklyn, Queens, and Staten Island**

### Map of Neighborhoods in NYC's Manhattan Borough <a name="neighbor-map"></a>

New York City has five boroughs:
    
    - Manhattan
    - The Bronx
    - Brooklyn
    - Queens
    - Staten Island
    
We will be looking more closely at neighborhoods in Manhattan, as this is where Broadway's Theatre District is located.

Let's take the Manhattan data and build that into a new dataframe for that borough.

In [296]:
manhattan_data_df = neighborhoods[neighborhoods['Borough'] == 'Manhattan'].reset_index(drop=True)
manhattan_data_df.head()

Borough        Neighborhood   Latitude  Longitude
0  Manhattan  Marble Hill         40.876551 -73.910660
1  Manhattan  Chinatown           40.715618 -73.994279
2  Manhattan  Washington Heights  40.851903 -73.936900
3  Manhattan  Inwood              40.867684 -73.921210
4  Manhattan  Hamilton Heights    40.823604 -73.949688

Let get the size of the Manhattan data frame:

In [297]:
manhattan_data_df.shape

(40, 4)

From our results, we can see there are 40 neighborhoods within Manhattan.

Let's take a look at all the values in the dataframe:

In [298]:
manhattan_data_df

Borough         Neighborhood   Latitude  Longitude
0   Manhattan  Marble Hill          40.876551 -73.910660
1   Manhattan  Chinatown            40.715618 -73.994279
2   Manhattan  Washington Heights   40.851903 -73.936900
3   Manhattan  Inwood               40.867684 -73.921210
4   Manhattan  Hamilton Heights     40.823604 -73.949688
5   Manhattan  Manhattanville       40.816934 -73.957385
6   Manhattan  Central Harlem       40.815976 -73.943211
7   Manhattan  East Harlem          40.792249 -73.944182
8   Manhattan  Upper East Side      40.775639 -73.960508
9   Manhattan  Yorkville            40.775930 -73.947118
10  Manhattan  Lenox Hill           40.768113 -73.958860
11  Manhattan  Roosevelt Island     40.762160 -73.949168
12  Manhattan  Upper West Side      40.787658 -73.977059
13  Manhattan  Lincoln Square       40.773529 -73.985338
14  Manhattan  Clinton              40.759101 -73.996119
15  Manhattan  Midtown              40.754691 -73.981669
16  Manhattan  Murray Hill          40.748303 -73.978332
17  Manhattan  Chelsea              40.744035 -74.003116
18  Manhattan  Greenwich Village    40.726933 -73.999914
19  Manhattan  East Village         40.727847 -73.982226
20  Manhattan  Lower East Side      40.717807 -73.980890
21  Manhattan  Tribeca              40.721522 -74.010683
22  Manhattan  Little Italy         40.719324 -73.997305
23  Manhattan  Soho                 40.722184 -74.000657
24  Manhattan  West Village         40.734434 -74.006180
25  Manhattan  Manhattan Valley     40.797307 -73.964286
26  Manhattan  Morningside Heights  40.808000 -73.963896
27  Manhattan  Gramercy             40.737210 -73.981376
28  Manhattan  Battery Park City    40.711932 -74.016869
29  Manhattan  Financial District   40.707107 -74.010665
30  Manhattan  Carnegie Hill        40.782683 -73.953256
31  Manhattan  Noho                 40.723259 -73.988434
32  Manhattan  Civic Center         40.715229 -74.005415
33  Manhattan  Midtown South        40.748510 -73.988713
34  Manhattan  Sutton Place         40.760280 -73.963556
35  Manhattan  Turtle Bay           40.752042 -73.967708
36  Manhattan  Tudor City           40.746917 -73.971219
37  Manhattan  Stuyvesant Town      40.731000 -73.974052
38  Manhattan  Flatiron             40.739673 -73.990947
39  Manhattan  Hudson Yards         40.756658 -74.000111

### Build map of Manhattan neighborhoods from latitude and longitude coordinates ###

Close to Times Square, there is a TKTS ticket booth that sells ticket to all the Broadway shows.

Let's set our address in Nominatim for the TKTS ticket booth location get the coordinates.

Let's draw a map using Folium, marking the TKTS booth near Times Square as a central location in the Theatre District, with green markers for the neighborhoods in Manhattan.

In [299]:
address = 'Broadway at, W 47th St, New York, NY 10036'

geolocator = Nominatim(user_agent="ny_explorer")
location = geolocator.geocode(address)
latitude = 40.7591855
longitude = -73.9848361
print('The geograpical coordinates of the TKTS booth near Times Square are {}, {}.'.format(latitude, longitude))

The geograpical coordinates of the TKTS booth near Times Square are 40.7591855, -73.9848361.


In [300]:
# create map of the Theatre District in Manhattan using latitude and longitude values
map_manhattan = folium.Map(location=[latitude, longitude], zoom_start=11)

# add a red circle marker to represent the TKTS ticket booth
folium.features.CircleMarker(
    [latitude, longitude],
    radius=5,
    color='red',
    popup='TKTS ticket booth in Times Square',
    fill = True,
    fill_color = 'red',
    fill_opacity = 0.6
).add_to(map_manhattan)

# add markers to map
for lat, lng, label in zip(manhattan_data['Latitude'], manhattan_data['Longitude'], manhattan_data['Neighborhood']):
    label = folium.Popup(label, parse_html=True)
    folium.CircleMarker(
        [lat, lng],
        radius=5,
        popup=label,
        color='green',
        fill=True,
        fill_color='#3186cc',
        fill_opacity=0.7,
        parse_html=False).add_to(map_manhattan)  
    
map_manhattan

**Manhattan, New York with markers for the 40 neighborhoods, and the TKTS booth near Times Square marked in red.**

## 3. Broadway Theatres and Current Productions <a name="broadway-3"></a>

## Theatres on Broadway - Maps and Analysis ##

Now we have some hotels in our dataset, let's have a look at the 40 theatres currently showing Broadway productions.

**Some of the most popular shows at the time of this project are**:
![Shows](shows.jpg)

Let's start by using BeautifulSoup to parse a list of the theatres and currently running shows from the following Wikipedia page:
https://en.wikipedia.org/wiki/Broadway_theatre

### Build pandas dataframe with Theatre Data scraped from a Broadway Wiki Page using BeautifulSoup

In [301]:
# Import Pandas and Beautiful Soup libraries
import pandas as pd
import urllib.request
from bs4 import BeautifulSoup
import requests

In [302]:
# Use Beautiful Soup to get the theatre data from the wiki page
sourcelink = 'https://en.wikipedia.org/wiki/Broadway_theatre'
source = requests.get(sourcelink).text
soup = BeautifulSoup(source, 'html.parser')
print(soup.prettify())

<!DOCTYPE html>
<html class="client-nojs" dir="ltr" lang="en">
 <head>
  <meta charset="utf-8"/>
  <title>
   Broadway theatre - Wikipedia
  </title>
  <script>
   document.documentElement.className="client-js";RLCONF={"wgBreakFrames":!1,"wgSeparatorTransformTable":["",""],"wgDigitTransformTable":["",""],"wgDefaultDateFormat":"dmy","wgMonthNames":["","January","February","March","April","May","June","July","August","September","October","November","December"],"wgMonthNamesShort":["","Jan","Feb","Mar","Apr","May","Jun","Jul","Aug","Sep","Oct","Nov","Dec"],"wgRequestId":"XjrEPgpAADoAAF-vfJ0AAAEG","wgCSPNonce":!1,"wgCanonicalNamespace":"","wgCanonicalSpecialPageName":!1,"wgNamespaceNumber":0,"wgPageName":"Broadway_theatre","wgTitle":"Broadway theatre","wgCurRevisionId":938745947,"wgRevisionId":938745947,"wgArticleId":725252,"wgIsArticle":!0,"wgIsRedirect":!1,"wgAction":"view","wgUserName":null,"wgUserGroups":["*"],"wgCategories":["Webarchive template wayback links","Articles with short de

In [303]:
# Build dataframe with theatre data

Theatre_Data_df = pd.DataFrame({'Theatre':'',
                          'Address':'',
                          'Capacity':int(),
                          'OwnerOperator':'',
                          'CurrentProduction':'',
                          'Type':'',
                          'Opening':'',
                          'Closing':'',
}, 
             index=[1])

In [304]:
# Build theatres table from matching table on the theatres wiki page and view the HTML code
theatres_table = soup.find('table', {'class':'wikitable sortable'})
theatres_table

<table class="wikitable sortable" style="font-size:96%;">
<tbody><tr>
<th>Theatre
</th>
<th>Address
</th>
<th>Capacity
</th>
<th>Owner/Operator
</th>
<th>Current production
</th>
<th>Type
</th>
<th>Opening
</th>
<th>Closing
</th></tr>
<tr>
<td><a href="/wiki/Al_Hirschfeld_Theatre" title="Al Hirschfeld Theatre">Al Hirschfeld Theatre</a></td>
<td>W. 45th St. <small>(No. 302)</small></td>
<td>1424</td>
<td><a href="/wiki/Jujamcyn_Theaters" title="Jujamcyn Theaters">Jujamcyn Theaters</a></td>
<td><i><a href="/wiki/Moulin_Rouge!_(musical)" title="Moulin Rouge! (musical)">Moulin Rouge!</a></i><sup class="reference" id="cite_ref-47"><a href="#cite_note-47">[46]</a></sup></td>
<td>Musical</td>
<td><span style="display:none">2019-07-25</span>July 25, 2019</td>
<td>Open-ended
</td></tr>
<tr>
<td><a href="/wiki/Ambassador_Theatre_(New_York_City)" title="Ambassador Theatre (New York City)">Ambassador Theatre</a></td>
<td>W. 49th St. <small>(No. 219)</small></td>
<td>1125</td>
<td><a href="/wiki/Th

### Loop through the theatres table and load our dataframe row by row.

In [305]:
# Initialize values for dataframe fields
Theatre = 0
Address = 0
Capacity = 0
OwnerOperator = 0
CurrentProduction = 0
Type = 0
Opening = 0
Closing = 0

for tr in theatres_table.find_all('tr'):
    i = 0
    for td in tr.find_all('td'):
        if i == 0:
            Theatre = td.text
            i = i + 1
        elif i == 1:
            Address = td.text
            i = i + 1
        elif i == 2:
            Capacity = td.text
            i = i + 1
        elif i == 3:
            Owner_Operator = td.text
            i = i + 1
        elif i == 4:
            CurrentProduction = td.text
            i = i + 1
        elif i == 5:
            Type = td.text
            i = i + 1
        elif i == 6:
            Opening = td.text
            i = i + 1
        elif i == 7:
            Closing = td.text.strip('\n').replace(']','')
            i = i + 1        
            Theatre_Data_df = Theatre_Data_df.append({'Theatre': Theatre,'Address': Address,'Capacity': Capacity, 'OwnerOperator': Owner_Operator, 'CurrentProduction': CurrentProduction, 'Type': Type, 'Opening': Opening, 'Closing': Closing},ignore_index=True)

In [306]:
# Let's look at the data in our theatres dataframe
Theatre_Data_df

Theatre                            Address Capacity  \
0                                                                    0         
1   Al Hirschfeld Theatre         W. 45th St. (No. 302)              1424      
2   Ambassador Theatre            W. 49th St. (No. 219)              1125      
3   American Airlines Theatre     W. 42nd St. (No. 227)              740       
4   August Wilson Theatre         W. 52nd St. (No. 245)              1228      
5   Belasco Theatre               W. 44th St. (No. 111)              1018      
6   Bernard B. Jacobs Theatre     W. 45th St. (No. 242)              1078      
7   Booth Theatre                 W. 45th St. (No. 222)              766       
8   Broadhurst Theatre            W. 44th St. (No. 235)              1186      
9   Broadway Theatre              W. 53rd St & Broadway (No. 1681)   1761      
10  Brooks Atkinson Theatre       W. 47th St. (No. 256)              1094      
11  Circle in the Square Theatre  W. 50th St. (No. 235)              840       
12  Cort Theatre                  W. 48th St. (No. 138)              1084      
13  Ethel Barrymore Theatre       W. 47th St. (No. 243)              1096      
14  Eugene O'Neill Theatre        W. 49th St. (No. 230)              1066      
15  Gerald Schoenfeld Theatre     W. 45th St. (No. 236)              1079      
16  Gershwin Theatre              W. 51st St. (No. 222)              1933      
17  Hayes Theater                 W. 44th St. (No. 240)              597       
18  Hudson Theatre                W. 44th St. (No. 141)              975       
19  Imperial Theatre              W. 45th St. (No. 249)              1443      
20  John Golden Theatre           W. 45th St. (No. 252)              805       
21  Longacre Theatre              W. 48th St. (No. 220)              1091      
22  Lunt-Fontanne Theatre         W. 46th St. (No. 205)              1519      
23  Lyceum Theatre                W. 45th St. (No. 149)              922       
24  Lyric Theatre                 W. 43rd St. (No. 214)              1622      
25  Majestic Theatre              W. 44th St. (No. 245)              1645      
26  Marquis Theatre               W. 46th St. (No. 210)              1612      
27  Minskoff Theatre              W. 45th St. (No. 200)              1710      
28  Music Box Theatre             W. 45th St. (No. 239)              1009      
29  Nederlander Theatre           W. 41st St. (No. 208)              1235      
30  Neil Simon Theatre            W. 52nd St. (No. 250)              1467      
31  New Amsterdam Theatre         W. 42nd St. (No. 214)              1747      
32  Richard Rodgers Theatre       W. 46th St. (No. 226)              1400      
33  St. James Theatre             W. 44th St. (No. 246)              1709      
34  Samuel J. Friedman Theatre    W. 47th St. (No. 261)              650       
35  Shubert Theatre               W. 44th St. (No. 225)              1460      
36  Stephen Sondheim Theatre      W. 43rd St. (No. 124)              1055      
37  Studio 54                     W. 54th St. (No. 254)              1006      
38  Vivian Beaumont Theater       W. 65th St. (No. 150)              1080      
39  Walter Kerr Theatre           W. 48th St. (No. 219)              945       
40  Winter Garden Theatre         W. 50th St. & Broadway (No. 1634)  1526      

                 OwnerOperator                    CurrentProduction     Type  \
0                                                                              
1   Jujamcyn Theaters           Moulin Rouge![46]                    Musical   
2   Shubert Organization        Chicago                              Musical   
3   Roundabout Theatre Company  A Soldier's Play[47]                 Play      
4   Jujamcyn Theaters           Mean Girls                           Musical   
5   Shubert Organization        Girl from the North Country[48]      Musical   
6   Shubert Organization        Company[49]                          Musical   
7   Shubert Organization

In [307]:
# Let's drop the empty first row
Theatre_Data_df.drop([0], inplace=True)

In [308]:
# Let's drop some of the columns that aren't needed
Theatre_Data_df.drop(['OwnerOperator','Type','Opening','Closing'],axis=1)
Theatre_Data_df

Theatre                            Address Capacity  \
1   Al Hirschfeld Theatre         W. 45th St. (No. 302)              1424      
2   Ambassador Theatre            W. 49th St. (No. 219)              1125      
3   American Airlines Theatre     W. 42nd St. (No. 227)              740       
4   August Wilson Theatre         W. 52nd St. (No. 245)              1228      
5   Belasco Theatre               W. 44th St. (No. 111)              1018      
6   Bernard B. Jacobs Theatre     W. 45th St. (No. 242)              1078      
7   Booth Theatre                 W. 45th St. (No. 222)              766       
8   Broadhurst Theatre            W. 44th St. (No. 235)              1186      
9   Broadway Theatre              W. 53rd St & Broadway (No. 1681)   1761      
10  Brooks Atkinson Theatre       W. 47th St. (No. 256)              1094      
11  Circle in the Square Theatre  W. 50th St. (No. 235)              840       
12  Cort Theatre                  W. 48th St. (No. 138)              1084      
13  Ethel Barrymore Theatre       W. 47th St. (No. 243)              1096      
14  Eugene O'Neill Theatre        W. 49th St. (No. 230)              1066      
15  Gerald Schoenfeld Theatre     W. 45th St. (No. 236)              1079      
16  Gershwin Theatre              W. 51st St. (No. 222)              1933      
17  Hayes Theater                 W. 44th St. (No. 240)              597       
18  Hudson Theatre                W. 44th St. (No. 141)              975       
19  Imperial Theatre              W. 45th St. (No. 249)              1443      
20  John Golden Theatre           W. 45th St. (No. 252)              805       
21  Longacre Theatre              W. 48th St. (No. 220)              1091      
22  Lunt-Fontanne Theatre         W. 46th St. (No. 205)              1519      
23  Lyceum Theatre                W. 45th St. (No. 149)              922       
24  Lyric Theatre                 W. 43rd St. (No. 214)              1622      
25  Majestic Theatre              W. 44th St. (No. 245)              1645      
26  Marquis Theatre               W. 46th St. (No. 210)              1612      
27  Minskoff Theatre              W. 45th St. (No. 200)              1710      
28  Music Box Theatre             W. 45th St. (No. 239)              1009      
29  Nederlander Theatre           W. 41st St. (No. 208)              1235      
30  Neil Simon Theatre            W. 52nd St. (No. 250)              1467      
31  New Amsterdam Theatre         W. 42nd St. (No. 214)              1747      
32  Richard Rodgers Theatre       W. 46th St. (No. 226)              1400      
33  St. James Theatre             W. 44th St. (No. 246)              1709      
34  Samuel J. Friedman Theatre    W. 47th St. (No. 261)              650       
35  Shubert Theatre               W. 44th St. (No. 225)              1460      
36  Stephen Sondheim Theatre      W. 43rd St. (No. 124)              1055      
37  Studio 54                     W. 54th St. (No. 254)              1006      
38  Vivian Beaumont Theater       W. 65th St. (No. 150)              1080      
39  Walter Kerr Theatre           W. 48th St. (No. 219)              945       
40  Winter Garden Theatre         W. 50th St. & Broadway (No. 1634)  1526      

                 OwnerOperator                    CurrentProduction     Type  \
1   Jujamcyn Theaters           Moulin Rouge![46]                    Musical   
2   Shubert Organization        Chicago                              Musical   
3   Roundabout Theatre Company  A Soldier's Play[47]                 Play      
4   Jujamcyn Theaters           Mean Girls                           Musical   
5   Shubert Organization        Girl from the North Country[48]      Musical   
6   Shubert Organization        Company[49]                          Musical   
7   Shubert Organization        Who's Afraid of Virginia Woolf?[50]  Play      
8   Shubert Organization        Jagged Little Pill[51]               Musical   
9   Shubert Organization

Let's download the data into a .csv file for manual processing in Excel. We will clean up the format of the address data.

In [309]:
Theatre_Data_df.to_csv(r'C:\Users\brian\BroadwayTheatres.csv')

In [310]:
# Build a new dataframe with cleaner theatre data
Column_Names = ['Theatre','Address','City','State','ZipCode','Latitude','Longitude','CurrentProduction','Type']
Theatre_Data_df4 = pd.DataFrame(columns=Column_Names)

In [311]:
Theatre_Data_df4 = pd.read_csv('BroadwayTheatres2.csv',dtype={'Latitude': float,'Longitude': float})
Theatre_Data_df4

Theatre                        Address      City  \
0   Al Hirschfeld Theatre         302 West 45th St.              New York   
1   Ambassador Theatre            219 West 49th St.              New York   
2   American Airlines Theatre     227 West 42nd St.              New York   
3   August Wilson Theatre         245 West 52nd St.              New York   
4   Belasco Theatre               111 West 44th St.              New York   
5   Bernard B. Jacobs Theatre     242 West 45th St.              New York   
6   Booth Theatre                 222 West 45th St.              New York   
7   Broadhurst Theatre            235 West 44th St.              New York   
8   Broadway Theatre              1681 West 53rd St & Broadway   New York   
9   Brooks Atkinson Theatre       256 West 47th St.              New York   
10  Circle in the Square Theatre  235 West 50th St.              New York   
11  Cort Theatre                  138 West 48th St.              New York   
12  Ethel Barrymore Theatre       243 West 47th St.              New York   
13  Eugene O'Neill Theatre        230 West 49th St.              New York   
14  Gerald Schoenfeld Theatre     236 West 45th St.              New York   
15  Gershwin Theatre              222 West 51st St.              New York   
16  Hayes Theater                 240 West 44th St.              New York   
17  Hudson Theatre                141 West 44th St.              New York   
18  Imperial Theatre              249 West 45th St.              New York   
19  John Golden Theatre           252 West 45th St.              New York   
20  Longacre Theatre              220 West 48th St.              New York   
21  Lunt-Fontanne Theatre         205 West 46th St.              New York   
22  Lyceum Theatre                149 West 45th St.              New York   
23  Lyric Theatre                 214 West 43rd St.              New York   
24  Majestic Theatre              245 West 44th St.              New York   
25  Marquis Theatre               210 West 46th St.              New York   
26  Minskoff Theatre              200 West 45th St.              New York   
27  Music Box Theatre             239 West 45th St.              New York   
28  Nederlander Theatre           208 West 41st St.              New York   
29  Neil Simon Theatre            250 West 52nd St.              New York   
30  New Amsterdam Theatre         214 West 42nd St.              New York   
31  Richard Rodgers Theatre       226 West 46th St.              New York   
32  St. James Theatre             246 West 44th St.              New York   
33  Samuel J. Friedman Theatre    261 West 47th St.              New York   
34  Shubert Theatre               225 West 44th St.              New York   
35  Stephen Sondheim Theatre      124 West 43rd St.              New York   
36  Studio 54                     254 West 54th St.              New York   
37  Vivian Beaumont Theater       150 West 65th St.              New York   
38  Walter Kerr Theatre           219 West 48th St.              New York   
39  Winter Garden Theatre         1634 West 50th St. & Broadway  New York   

   State  Capacity                    CurrentProduction     Type   Latitude  \
0   NY    1424      Moulin Rouge![46]                    Musical  40.759253   
1   NY    1125      Chicago                              Musical  40.761236   
2   NY    740       A Soldier's Play[47]                 Play     40.757156   
3   NY    1228      Mean Girls                           Musical  40.763373   
4   NY    1018      Girl from the North Country[48]      Musical  40.756644   
5   NY    1078      Company[49]                          Musical  40.758608   
6   NY    766       Who's Afraid of Virginia Woolf?[50]  Play     40.758373   
7   NY    1186      Jagged Little Pill[51]               Musical  40.758269   
8   NY    1761      West Side Story[52]                  Musical  40.764457   
9   NY    1094      Six[53]                              Musical  40.759975  

### Map of Broadway Theatre Locations <a name="broadway-map"></a>

In [312]:
# Setting the latitude and longitude of Broadway

broadway_latitude = 40.75659 # neighborhood latitude value
broadway_longitude = -73.98626, # neighborhood longitude value

neighborhood_name = 'Broadway'

print('Latitude and longitude values of {} are {}, {}.'.format(neighborhood_name, 
                                                               broadway_latitude, 
                                                               broadway_longitude))

Latitude and longitude values of Broadway are 40.75659, (-73.98626,).


In [313]:
# create map of the Broadway theatre district using latitude and longitude values
map_broadway = folium.Map(location=[broadway_latitude, broadway_longitude], zoom_start=14)

# add markers to map
for lat, lng, label in zip(Theatre_Data_df4['Latitude'], Theatre_Data_df4['Longitude'], Theatre_Data_df4['Theatre']):
    label = folium.Popup(label, parse_html=True)
    folium.CircleMarker(
        [lat, lng],
        radius=5,
        popup=label,
        color='black',
        fill=True,
        fill_color='purple',
        fill_opacity=0.7,
        parse_html=False).add_to(map_broadway) 

map_broadway

**Map Showing Locations of 40 Broadway Theatres in Manhattan, New York City**

#### Theatre Location Analysis ####

Looking at the mapped results, we see that the greatest concentration of theatres appears clustered near 44th and 45th streets between 7th Avenue and 8th Avenue.

This area is looking like a great spot to find local restaurants and hotels.

Let's set our starting point at the Richard Rodgers Theatre where *Hamilton* is currently playing.

This theatre is also located near the center of our cluster:

![Richard Rodgers Theatre](RichardRodgersMap.jpg)


**Google Maps street level view of the intersection of West 47th Street and 7th Avenue from the TKTS booth location**:
![TKTS Booth View](TKTSView.jpg)

## 4. Hotels Near the Broadway Theatre District <a name="hotels-4"></a>

### Search with the Foursquare API for Hotels near the Richard Rodgers Theatre

We will use the address of Richard Rodgers Theatre as our starting point as we look for nearby hotels using the Foursquare API.

Let's get the hotel's latitude and longitude values.

In [314]:
# Set address of Richard Rodgers Theatre as our starting point
address = '226 West 46th Street, New York, NY'
geolocator = Nominatim(user_agent="foursquare_agent")
location = geolocator.geocode(address)
latitude = location.latitude
longitude = location.longitude
print(latitude, longitude)
    

40.7590309 -73.98674788284762


We can use Foursquare to view hotels within a given radius of the Richard Rodgers Theatre.

In [315]:
search_query = 'Hotel'
radius = 500
print(search_query + ' .... OK!')

Hotel .... OK!


In [316]:
url = 'https://api.foursquare.com/v2/venues/search?client_id={}&client_secret={}&ll={},{}&v={}&query={}&radius={}&limit={}'.format(CLIENT_ID, CLIENT_SECRET, latitude, longitude, VERSION, search_query, radius, LIMIT)
url

'https://api.foursquare.com/v2/venues/search?client_id=QQ5ZYS0QBI2WYIMIAZZUFEZJEV10V25IVVCFTW3EB550JISX&client_secret=3CJR0S30CC0RXHCOYTX5XUG1RCMVGA5MKV00WCOEQD41ESE1&ll=40.7590309,-73.98674788284762&v=20180604&query=Hotel&radius=500&limit=30'

Let's get the JSON data and set that into a variable called 'results'

In [317]:
results = requests.get(url).json()
results

{'meta': {'code': 200, 'requestId': '5e3c5d591a4b0a001b6397da'},
 'response': {'venues': [{'id': '4b4bbe3ff964a52016a626e3',
    'name': 'Hotel Edison',
    'location': {'address': '228 W 47th St',
     'crossStreet': 'at 8th Ave',
     'lat': 40.75966639038906,
     'lng': -73.98608770066558,
     'labeledLatLngs': [{'label': 'display',
       'lat': 40.75966639038906,
       'lng': -73.98608770066558}],
     'distance': 90,
     'postalCode': '10036',
     'cc': 'US',
     'city': 'New York',
     'state': 'NY',
     'country': 'United States',
     'formattedAddress': ['228 W 47th St (at 8th Ave)',
      'New York, NY 10036',
      'United States']},
    'categories': [{'id': '4bf58dd8d48988d1fa931735',
      'name': 'Hotel',
      'pluralName': 'Hotels',
      'shortName': 'Hotel',
      'icon': {'prefix': 'https://ss3.4sqi.net/img/categories_v2/travel/hotel_',
       'suffix': '.png'},
      'primary': True}],
    'venuePage': {'id': '501560524'},
    'referralId': 'v-1581014290',

In [318]:
# assign relevant part of JSON to venues
venues = results['response']['venues']

# tranform venues into a dataframe
dataframe = json_normalize(venues)
dataframe.head()

categories  \
0  [{'id': '4bf58dd8d48988d1fa931735', 'name': 'Hotel', 'pluralName': 'Hotels', 'shortName': 'Hotel', 'icon': {'prefix': 'https://ss3.4sqi.net/img/categories_v2/travel/hotel_', 'suffix': '.png'}, 'primary': True}]   
1  [{'id': '4bf58dd8d48988d1fa931735', 'name': 'Hotel', 'pluralName': 'Hotels', 'shortName': 'Hotel', 'icon': {'prefix': 'https://ss3.4sqi.net/img/categories_v2/travel/hotel_', 'suffix': '.png'}, 'primary': True}]   
2  [{'id': '4bf58dd8d48988d1fa931735', 'name': 'Hotel', 'pluralName': 'Hotels', 'shortName': 'Hotel', 'icon': {'prefix': 'https://ss3.4sqi.net/img/categories_v2/travel/hotel_', 'suffix': '.png'}, 'primary': True}]   
3  [{'id': '4bf58dd8d48988d1fa931735', 'name': 'Hotel', 'pluralName': 'Hotels', 'shortName': 'Hotel', 'icon': {'prefix': 'https://ss3.4sqi.net/img/categories_v2/travel/hotel_', 'suffix': '.png'}, 'primary': True}]   
4  [{'id': '4bf58dd8d48988d1fa931735', 'name': 'Hotel', 'pluralName': 'Hotels', 'shortName': 'Hotel', 'icon': {'prefix': 'https://ss3.4sqi.net/img/categories_v2/travel/hotel_', 'suffix': '.png'}, 'primary': True}]   

  delivery.id delivery.provider.icon.name delivery.provider.icon.prefix  \
0  NaN         NaN                         NaN                            
1  NaN         NaN                         NaN                            
2  NaN         NaN                         NaN                            
3  NaN         NaN                         NaN                            
4  NaN         NaN                         NaN                            

  delivery.provider.icon.sizes delivery.provider.name delivery.url  hasPerk  \
0  NaN                          NaN                    NaN          False     
1  NaN                          NaN                    NaN          False     
2  NaN                          NaN                    NaN          False     
3  NaN                          NaN                    NaN          False     
4  NaN                          NaN                    NaN          False     

                         id location.address location.cc location.city  \
0  4b4bbe3ff964a52016a626e3  228 W 47th St    US          New York       
1  4d7ad1263fbf6dcbb60b6423  132 W 47th St    US          New York       
2  5bc65d8e1822230025fffcb1  310 W 40th St    US          New York       
3  4adbaf34f964a520012a21e3  235 W 46th St    US          New York       
4  5093c236830214706abb75db  218 W 50th St    US          New York       

  location.country location.crossStreet  location.distance  \
0  United States    at 8th Ave           90                  
1  United States    btwn 6th & 7th Ave.  303                 
2  United States    NaN                  514                 
3  United States    NaN                  66                  
4  United States    50th & B'way         332                 

                                                  location.formattedAddress  \
0  [228 W 47th St (at 8th Ave), New York, NY 10036, United States]            
1  [132 W 47th St (btwn 6th & 7th Ave.), New York, NY 10036, United States]   
2  [310 W 40th St, New York, NY 10018, United States]                         
3  [235 W 46th St, New York, NY 10036, United States]                         
4  [218 W 50th St (50th & B'way), New York, NY 10019, United States]          

                                                        location.labeledLatLngs  \
0  [{'label': 'display', 'lat': 40.75966639038906, 'lng': -73.98608770066558}]    
1  [{'label': 'display', 'lat': 40.7583581, 'lng': -73.9832623}]                  
2  [{'label': 'display', 'lat': 40.75620715052, 'lng': -73.99157595205548}]       
3  [{'label': 'display', 'lat': 40.759568887065484, 'lng': -73.98709530548786}]   
4  [{'label': 'display', 'lat': 40.761691026346256, 'lng': -73.98495316332877}]   

   location.lat  location.lng location.neighborhood location.postalCode  \
0  40.759666    -73.986088     NaN                   10036                
1  40.758358    -73.983262     N

#### Let's look at data for Hotels within 500 meters of the Richard Rodgers Theatre

In [319]:
# keep only columns that include venue name, and anything that is associated with location
filtered_columns2 = ['name', 'categories'] + [col for col in dataframe.columns if col.startswith('location.')] + ['id']
dataframe_filtered2 = dataframe.loc[:, filtered_columns2]

# function that extracts the category of the venue
def get_category_type(row):
    try:
        categories_list = row['categories']
    except:
        categories_list = row['venue.categories']
        
    if len(categories_list) == 0:
        return None
    else:
        return categories_list[0]['name']

# filter the category for each row
dataframe_filtered2['categories'] = dataframe_filtered2.apply(get_category_type, axis=1)

# clean column names by keeping only last term
dataframe_filtered2.columns = [column.split('.')[-1] for column in dataframe_filtered2.columns]

dataframe_filtered2

name  \
0   Hotel Edison                                                         
1   Sanctuary Hotel New York                                             
2   Aliz Hotel                                                           
3   Paramount Hotel                                                      
4   citizenM Hotel New York Times Square                                 
5   The Algonquin Hotel, Autograph Collection                            
6   The Manhattan at Times Square Hotel                                  
7   Kimpton Muse Hotel                                                   
8   Millennium Broadway Hotel                                            
9   Renaissance New York Times Square Hotel                              
10  Royalton Hotel                                                       
11  The Belvedere Hotel                                                  
12  The Pearl Hotel                                                      
13  Night Hotel Times Square                                             
14  AC Hotel Times Square                                                
15  Distrikt Hotel                                                       
16  The Michelangelo Hotel                                               
17  Serafina Time Hotel                                                  
18  Blue Bar at The Algonquin Hotel Times Square, Autograph Collection   
19  The Hotel @ Times Square                                             
20  Night Hotel                                                          
21  Hotel MELA                                                           
22  Room Mate Grace Hotel                                                
23  Merrion Row Hotel and Public House                                   
24  Cassa Hotel NY 45th Street                                           
25  Hotel St James                                                       
26  Amsterdam Court Hotel                                                
27  The Muse Hotel Fitness Center                                        
28  The Premier Hotel New York                                           
29  Riu Plaza Times Square Hotel                                         

            categories  \
0   Hotel                
1   Hotel                
2   Hotel                
3   Hotel                
4   Hotel                
5   Hotel                
6   Hotel                
7   Hotel                
8   Hotel                
9   Hotel                
10  Hotel                
11  Hotel                
12  Hotel                
13  Hotel                
14  Hotel                
15  Hotel                
16  Hotel                
17  Italian Restaurant   
18  Hotel Bar            
19  Hotel                
20  Hotel                
21  Hotel                
22  Hotel                
23  Hotel                
24  Hotel                
25  Hotel                
26  Hotel                
27  Gym                  
28  Hotel                
29  Resort               

                                                   address  cc      city  \
0   228 W 47th St                                           US  New York   
1   132 W 47th St                                           US  New York   
2   310 W 40th St                                           US  New York   
3   235 W 46th St                                           US  New York   
4   218 W 50th St                                           US  New York   
5   59 W 44th St                                            US  New York   
6   790 7th Ave                                             US  New York   
7   130 W 46th St                                           US  New York   
8   145 W 44th St                                           US  New York   
9   Two Times Square, 714 Seventh Avenue At W. 48th Street  US  New York   
10  44 W 44th St                                            US  New York   
11  319 W 48th St                                   

In [320]:
dataframe_filtered2.name

0     Hotel Edison                                                      
1     Sanctuary Hotel New York                                          
2     Aliz Hotel                                                        
3     Paramount Hotel                                                   
4     citizenM Hotel New York Times Square                              
5     The Algonquin Hotel, Autograph Collection                         
6     The Manhattan at Times Square Hotel                               
7     Kimpton Muse Hotel                                                
8     Millennium Broadway Hotel                                         
9     Renaissance New York Times Square Hotel                           
10    Royalton Hotel                                                    
11    The Belvedere Hotel                                               
12    The Pearl Hotel                                                   
13    Night Hotel Times Square                     

Let's map out the hotels returned in our Foursquare search.

### Map of Hotels Near Broadway Theatres <a name="hotels-map"></a>

First let's show a map of the hotel locations, so we can get a view of where they are in relation to the theatre.

In [321]:
# create map of Manhattan using latitude and longitude values
map_hotels = folium.Map(location=[latitude, longitude], zoom_start=15)

# add a red circle marker to represent the Midtown Manhattan neighborhood center
folium.features.CircleMarker(
    [latitude, longitude],
    radius=5,
    color='red',
    popup='Richard Rodgers Theatre - Home of Hamilton',
    fill = True,
    fill_color = 'red',
    fill_opacity = 0.6
).add_to(map_hotels)

# add markers to map
for lat, lng, label in zip(dataframe_filtered2['lat'], dataframe_filtered2['lng'], dataframe_filtered2['name']):
    label = folium.Popup(label, parse_html=True)
    folium.CircleMarker(
        [lat, lng],
        radius=5,
        popup=label,
        color='purple',
        fill=True,
        fill_color='purple',
        fill_opacity=0.5,
        parse_html=False).add_to(map_hotels)  
    
map_hotels

**Map of Hotels Near the Richard Rodgers Theatre returned by searching for Hotels within a 500 meter radius**

Let's sort the data for the hotels to view the closest ones from the theater, then list the others in ascending order.

In [322]:
dataframe_filtered2.sort_values(by=['distance'])

name  \
3   Paramount Hotel                                                      
0   Hotel Edison                                                         
29  Riu Plaza Times Square Hotel                                         
9   Renaissance New York Times Square Hotel                              
13  Night Hotel Times Square                                             
8   Millennium Broadway Hotel                                            
17  Serafina Time Hotel                                                  
12  The Pearl Hotel                                                      
7   Kimpton Muse Hotel                                                   
20  Night Hotel                                                          
1   Sanctuary Hotel New York                                             
22  Room Mate Grace Hotel                                                
11  The Belvedere Hotel                                                  
27  The Muse Hotel Fitness Center                                        
28  The Premier Hotel New York                                           
4   citizenM Hotel New York Times Square                                 
23  Merrion Row Hotel and Public House                                   
26  Amsterdam Court Hotel                                                
21  Hotel MELA                                                           
25  Hotel St James                                                       
16  The Michelangelo Hotel                                               
6   The Manhattan at Times Square Hotel                                  
14  AC Hotel Times Square                                                
18  Blue Bar at The Algonquin Hotel Times Square, Autograph Collection   
24  Cassa Hotel NY 45th Street                                           
5   The Algonquin Hotel, Autograph Collection                            
19  The Hotel @ Times Square                                             
2   Aliz Hotel                                                           
10  Royalton Hotel                                                       
15  Distrikt Hotel                                                       

            categories  \
3   Hotel                
0   Hotel                
29  Resort               
9   Hotel                
13  Hotel                
8   Hotel                
17  Italian Restaurant   
12  Hotel                
7   Hotel                
20  Hotel                
1   Hotel                
22  Hotel                
11  Hotel                
27  Gym                  
28  Hotel                
4   Hotel                
23  Hotel                
26  Hotel                
21  Hotel                
25  Hotel                
16  Hotel                
6   Hotel                
14  Hotel                
18  Hotel Bar            
24  Hotel                
5   Hotel                
19  Hotel                
2   Hotel                
10  Hotel                
15  Hotel                

                                                   address  cc      city  \
3   235 W 46th St                                           US  New York   
0   228 W 47th St                                           US  New York   
29  W 46th Street                                           US  New York   
9   Two Times Square, 714 Seventh Avenue At W. 48th Street  US  New York   
13  157 W 47th St                                           US  New York   
8   145 W 44th St                                           US  New York   
17  224 W 49th St                                           US  New York   
12  233 W 49th St                                           US  New York   
7   130 W 46th St                                           US  New York   
20  132 W 45th St                                           US  New York   
1   132 W 47th St                                           US  New York   
22  125 W 45th St                                   

Let's save the query results to a local .CSV file.

Let's check the ratings of some the hotels in our dataset, starting with the closest one, the Paramount.

In [323]:
venue_id = '4adbaf34f964a520012a21e3' # Paramount Hotel ID
url = 'https://api.foursquare.com/v2/venues/{}?client_id={}&client_secret={}&v={}'.format(venue_id, CLIENT_ID, CLIENT_SECRET, VERSION) 

In [324]:
result = requests.get(url).json()
print(result['response']['venue'].keys())
result['response']['venue']

dict_keys(['id', 'name', 'contact', 'location', 'canonicalUrl', 'categories', 'verified', 'stats', 'url', 'hasMenu', 'likes', 'dislike', 'ok', 'rating', 'ratingColor', 'ratingSignals', 'menu', 'allowMenuUrlEdit', 'beenHere', 'specials', 'photos', 'venuePage', 'reasons', 'description', 'storeId', 'page', 'hereNow', 'createdAt', 'tips', 'shortUrl', 'timeZone', 'listed', 'hours', 'pageUpdates', 'inbox', 'attributes', 'bestPhoto', 'colors'])


{'id': '4adbaf34f964a520012a21e3',
 'name': 'Paramount Hotel',
 'contact': {'phone': '2127645500',
  'formattedPhone': '(212) 764-5500',
  'twitter': 'nycparamount',
  'facebook': '101786016079',
  'facebookUsername': 'NYCParamount',
  'facebookName': 'Paramount Hotel Times Square New York'},
 'location': {'address': '235 W 46th St',
  'lat': 40.759568887065484,
  'lng': -73.98709530548786,
  'labeledLatLngs': [{'label': 'display',
    'lat': 40.759568887065484,
    'lng': -73.98709530548786}],
  'postalCode': '10036',
  'cc': 'US',
  'city': 'New York',
  'state': 'NY',
  'country': 'United States',
  'formattedAddress': ['235 W 46th St',
   'New York, NY 10036',
   'United States']},
 'canonicalUrl': 'https://foursquare.com/v/paramount-hotel/4adbaf34f964a520012a21e3',
 'categories': [{'id': '4bf58dd8d48988d1fa931735',
   'name': 'Hotel',
   'pluralName': 'Hotels',
   'shortName': 'Hotel',
   'icon': {'prefix': 'https://ss3.4sqi.net/img/categories_v2/travel/hotel_',
    'suffix': '.pn

In [325]:
try:
    print(result['response']['venue']['rating'])
except:
    print('This venue has not been rated yet.')

6.1


That's not a very good rating, so let's try the citizenM Hotel which is only 332 meters away.

In [326]:
venue_id = '5093c236830214706abb75db' # citizenM Hotel ID
url = 'https://api.foursquare.com/v2/venues/{}?client_id={}&client_secret={}&v={}'.format(venue_id, CLIENT_ID, CLIENT_SECRET, VERSION) 

In [327]:
result = requests.get(url).json()
print(result['response']['venue'].keys())
result['response']['venue']

dict_keys(['id', 'name', 'contact', 'location', 'canonicalUrl', 'categories', 'verified', 'stats', 'url', 'likes', 'dislike', 'ok', 'rating', 'ratingColor', 'ratingSignals', 'allowMenuUrlEdit', 'beenHere', 'specials', 'photos', 'reasons', 'storeId', 'page', 'hereNow', 'createdAt', 'tips', 'shortUrl', 'timeZone', 'listed', 'pageUpdates', 'inbox', 'attributes', 'bestPhoto', 'colors'])


{'id': '5093c236830214706abb75db',
 'name': 'citizenM Hotel New York Times Square',
 'contact': {'phone': '2124613638',
  'formattedPhone': '(212) 461-3638',
  'twitter': 'citizenm',
  'facebook': '173364980699',
  'facebookUsername': 'citizenMhotels',
  'facebookName': 'citizenM hotels'},
 'location': {'address': '218 W 50th St',
  'crossStreet': "50th & B'way",
  'lat': 40.761691026346256,
  'lng': -73.98495316332877,
  'labeledLatLngs': [{'label': 'display',
    'lat': 40.761691026346256,
    'lng': -73.98495316332877}],
  'postalCode': '10019',
  'cc': 'US',
  'city': 'New York',
  'state': 'NY',
  'country': 'United States',
  'formattedAddress': ["218 W 50th St (50th & B'way)",
   'New York, NY 10019',
   'United States']},
 'canonicalUrl': 'https://foursquare.com/v/citizenm-hotel-new-york-times-square/5093c236830214706abb75db',
 'categories': [{'id': '4bf58dd8d48988d1fa931735',
   'name': 'Hotel',
   'pluralName': 'Hotels',
   'shortName': 'Hotel',
   'icon': {'prefix': 'https:/

In [328]:
try:
    print(result['response']['venue']['rating'])
except:
    print('This venue has not been rated yet.')

9.2


Wow, this looks like a great choice for our customers. Let's get the number of ratings for this hotel.

In [329]:
result['response']['venue']['ratingSignals']

412

Now let's get the number of tips.

That's a lot of ratings, so this seems like a good hotel for our customers. Let's get the hotel's tips from Foursquare.

In [330]:
result['response']['venue']['tips']['count']

83

In [331]:
tips = results['response']['tips']['items']

tip = results['response']['tips']['items'][0]
tip.keys()

KeyError: 'tips'

In [ ]:
pd.set_option('display.max_colwidth', -1)

tips_df = json_normalize(tips) # json normalize tips

# columns to keep
filtered_columns = ['text', 'agreeCount', 'disagreeCount', 'id', 'user.firstName', 'user.lastName', 'user.gender', 'user.id']
tips_filtered = tips_df.loc[:, filtered_columns]

# display tips
tips_filtered

Let's select the citizenM Hotel as our preferred hotel, and search for the best restaurants nearby.

## 5. Restaurants Near the Broadway Theatre District and citizenM Hotel <a name="restaurants-5"></a>

In [ ]:
# Latitude and Longitude of citizenM Hotel
latitude = 40.761691
longitude = -73.984953

In [ ]:
search_query = 'Restaurant'
radius = 500
print(search_query + ' .....OK!')

In [ ]:
# Set up the search url to query Foursquare
url = 'https://api.foursquare.com/v2/venues/search?client_id={}&client_secret={}&ll={},{}&v={}&query={}&radius={}&limit={}'.format(CLIENT_ID, CLIENT_SECRET, latitude, longitude, VERSION, search_query, radius, LIMIT)

In [ ]:
# Get the results as a JSON file
results = requests.get(url).json()
results

In [ ]:
# assign relevant part of JSON to venues
venues = results['response']['venues']

# tranform venues into a dataframe
dataframe = json_normalize(venues)
dataframe.head()

In [ ]:
# keep only columns that include venue name, and anything that is associated with location
filtered_columns = ['name', 'categories'] + [col for col in dataframe.columns if col.startswith('location.')] + ['id']
restaurants_filtered_df = dataframe.loc[:, filtered_columns]

# function that extracts the category of the venue
def get_category_type(row):
    try:
        categories_list = row['categories']
    except:
        categories_list = row['venue.categories']
        
    if len(categories_list) == 0:
        return None
    else:
        return categories_list[0]['name']

# filter the category for each row
restaurants_filtered_df['categories'] = restaurants_filtered_df.apply(get_category_type, axis=1)

# clean column names by keeping only last term
restaurants_filtered_df.columns = [column.split('.')[-1] for column in dataframe_filtered.columns]

restaurants_filtered_df

In [ ]:
restaurants_filtered_df.name

### Map of Restaurants Near the Broadway Theatre District <a name="rest-map"></a>

Let's map out the location of the 29 restaurants near the citizenM Hotel returned by our Foursquare query.

In [332]:
restaurants_map = folium.Map(location=[latitude, longitude], zoom_start=15) # generate map centred around the citizenM Hotel

# add a red circle marker to represent the citizenM Hotel
folium.features.CircleMarker(
    [latitude, longitude],
    radius=5,
    color='red',
    popup='citizenM Hotel',
    fill = True,
    fill_color = 'red',
    fill_opacity = 0.6
).add_to(restaurants_map)

# add the restaurants as blue circle markers
for lat, lng, name in zip(restaurants_filtered_df['lat'],restaurants_filtered_df['lng'],restaurants_filtered_df['name']):
    label = '{}'.format(name)
    label = folium.Popup(label, parse_html=True)
    folium.CircleMarker(
        [lat, lng],
        radius=5,
    popup=label,
    color='yellow',
    fill=True,
    fill_color='red',
    fill_opacity=0.7,
    parse_html=False).add_to(restaurants_map)

# display map
restaurants_map

**Restaurants within 500 meters of the citizenM Hotel, marked in red, returned by our Foursquare API query**

In [ ]:
Now that we have some nearby restaurants we can build out the details of our tour package.

## 6. Conclusions and Findings <a name="conclusions"></a>

We have researched the Broadway Theatre District in the Manhattan borough of New York City, and now have list of current shows and theatres where they are appearing.

We also have a list of hotels near the Richard Rodgers Theatre where Hamilton is continuing it's long rung. From this list, we have a recommended hotel, the citizenM with a 9.2 rating from over 400 users.

Finally, we have a list of restaurants close to our hotel we can build in our travel package.

Here is a report on the findings.

- [Current Broadway Productions and Associated Theatres](#current-productions)
- [Recommended Hotel](#recommend-hotel)
- [Nearby Restaurants](#nearby-rest)

### Current Broadway Productions and Associated Theatres <a name="current-productions"></a>

In [334]:
CurrentProductions_df = pd.read_csv('current_productions.csv')
CurrentProductions_df

Theatre                        Address  Capacity  \
0   Al Hirschfeld Theatre         302 West 45th St.              1424       
1   Ambassador Theatre            219 West 49th St.              1125       
2   American Airlines Theatre     227 West 42nd St.              740        
3   August Wilson Theatre         245 West 52nd St.              1228       
4   Belasco Theatre               111 West 44th St.              1018       
5   Bernard B. Jacobs Theatre     242 West 45th St.              1078       
6   Booth Theatre                 222 West 45th St.              766        
7   Broadhurst Theatre            235 West 44th St.              1186       
8   Broadway Theatre              1681 West 53rd St & Broadway   1761       
9   Brooks Atkinson Theatre       256 West 47th St.              1094       
10  Circle in the Square Theatre  235 West 50th St.              840        
11  Cort Theatre                  138 West 48th St.              1084       
12  Ethel Barrymore Theatre       243 West 47th St.              1096       
13  Eugene O'Neill Theatre        230 West 49th St.              1066       
14  Gerald Schoenfeld Theatre     236 West 45th St.              1079       
15  Gershwin Theatre              222 West 51st St.              1933       
16  Hayes Theater                 240 West 44th St.              597        
17  Hudson Theatre                141 West 44th St.              975        
18  Imperial Theatre              249 West 45th St.              1443       
19  John Golden Theatre           252 West 45th St.              805        
20  Longacre Theatre              220 West 48th St.              1091       
21  Lunt-Fontanne Theatre         205 West 46th St.              1519       
22  Lyceum Theatre                149 West 45th St.              922        
23  Lyric Theatre                 214 West 43rd St.              1622       
24  Majestic Theatre              245 West 44th St.              1645       
25  Minskoff Theatre              200 West 45th St.              1710       
26  Music Box Theatre             239 West 45th St.              1009       
27  Nederlander Theatre           208 West 41st St.              1235       
28  Neil Simon Theatre            250 West 52nd St.              1467       
29  New Amsterdam Theatre         214 West 42nd St.              1747       
30  Richard Rodgers Theatre       226 West 46th St.              1400       
31  St. James Theatre             246 West 44th St.              1709       
32  Samuel J. Friedman Theatre    261 West 47th St.              650        
33  Shubert Theatre               225 West 44th St.              1460       
34  Stephen Sondheim Theatre      124 West 43rd St.              1055       
35  Studio 54                     254 West 54th St.              1006       
36  Vivian Beaumont Theater       150 West 65th St.              1080       
37  Walter Kerr Theatre           219 West 48th St.              945        
38  Winter Garden Theatre         1634 West 50th St. & Broadway  1526       

                    CurrentProduction     Type  
0   Moulin Rouge                       Musical  
1   Chicago                            Musical  
2   A Soldier's Play                   Play     
3   Mean Girls                         Musical  
4   Girl from the North Country        Musical  
5   Company                            Musical  
6   Who's Afraid of Virginia Woolf?    Play     
7   Jagged Little Pill                 Musical  
8   West Side Story                    Musical  
9   Six                                Musical  
10  American Buffalo                   Play     
11  The Minutes                        Play     
12  The Inheritance                    Play     
13  The Book of Mormon                 Musical  
14  Come from Away                     Musical  
15  Wicked                             Musical  
16  Grand Horizons                     Play     
17  David Byrne's American Utopia      Special  
18  Ain't Too

### Recommended Hotel <a name="recommend-hotel"></a>

The recommended hotel for our package is the 4-Star citizenM Hotel.

![citizenM](citizenM.jpg)

- **Address**: 218 W 50th St, New York, NY 10019
- **Reviews**: 4.6 stars, with 1,587 reviews on Google.
- **Room Rates**: as low as $169 a night
- **Features**: In walking distance of many of the Broadway Theatres, this hotel has the following amenities:

**From the citizenM website**:

Located just off the corner of Broadway and 50th street, our New York hotel is right in the heart of Midtown Manhattan, smack bang in the middle of the Theater District. A 5 minute walk from 5th Avenue, Madison Avenue, Central Park and Columbus Circle, 10-15 minutes away from the major museums, Rockefeller Center and the Empire State Building, and an easy subway or cab ride to anywhere else in the city, from Soho to the Meatpacking District – so if you’re looking for the coolest hotels New York has to offer, check out citizenM.

### Nearby Restaurants <a name="nearby-rest"></a>

The following 30 restaurants are within walking distance of the citizenM Hotel:

In [333]:
Restaurants_df4 = pd.read_csv('BroadwayRestClean.csv', index_col=False)
Restaurants_df4

Restaurant                 Category  \
0   The View Restaurant & Lounge                  Restaurant                
1   Junior's Restaurant & Bakery                  American Restaurant       
2   Carmine's Italian Restaurant                  Italian Restaurant        
3   L'ybane Restaurant                            Cocktail Bar              
4   Smith's Bar & Restaurant                      Bar                       
5   Ding BBQ and Hot Pot Restaurant               Hotpot Restaurant         
6   Restaurant Row                                Other Great Outdoors      
7   O'Donoghues Pub & Restaurant                  Bar                       
8   Junior's Restaurant                           American Restaurant       
9   Connolly's Pub & Restaurant                   Irish Pub                 
10  Utsav Restaurant                              Indian Restaurant         
11  The Round Table Restaurant, at The Algonquin  American Restaurant       
12  Jack's Restaurant and Bar                     American Restaurant       
13  Restaurant Triomphe                           French Restaurant         
14  Via Italia Restaurant                         Italian Restaurant        
15  Chazz Palminteri Italian Restaurant           Italian Restaurant        
16  Subway Restaurant                             Sandwich Place            
17  Buenavista Restaurant & Bar                   Restaurant                
18  Tulcingo Del Valle Restaurant                 Mexican Restaurant        
19  Cancun Mexican Restaurant                     Mexican Restaurant        
20  PJ Moran's Irish Pub & Restaurant             Pub                       
21  Connolly's Pub & Restaurant                   Irish Pub                 
22  Patsy's Italian Restaurant                    Italian Restaurant        
23  Astro Restaurant                              Diner                     
24  Connolly's Pub & Restaurant                   Pub                       
25  Morning Star Restaurant                       American Restaurant       
26  The Flame Restaurant                          Diner                     
27  SOUTHGATE Bar & Restaurant                    American Restaurant       
28  McGarry's Bar & Restaurant                    Bar                       
29  The Garden Restaurant                         New American Restaurant   

                               Address  Distance  
0   1535 Broadway                       54        
1   1515 Broadway                       78        
2   200 W 44th St                       170       
3   709 8th Ave                         176       
4   701 8th Ave                         188       
5   234 W 48th St                       194       
6   W. 46th St.                         254       
7   156 W 44th St                       259       
8   1626 Broadway                       279       
9   121 W 45th St                       324       
10  1185 Avenue of the Americas         364       
11  59 W 44th St                        482       
12  147 W 40th St                       492       
13  49 W 44th St                        524       
14  45 W 46th St                        546       
15  30 W 46th St                        594       
16  31 W 43rd St                        600       
17  536 9th Ave                         619       
18  665 10th Ave (btwn 46th & 47th St)  683       
19  937 8th Ave                         782       
20  3 E 48th St                         799       
21  14 E 47th St                        817       
22  236 W 56th St                       819       
23  1361 6th Ave                        830       
24  43 W 54th St                        849       
25  879 9th Ave                         1011      
26  893 9th Ave                         1054      
27  154 Central Park S                  1061      
28  417 9th Ave                         1070      
29  57 E 57th St                        1324